src/mem/mem_ctrl.cc

```diff
@@ -939,11 +1681,12 @@ MemCtrl::processNextReqEvent()
         // track if we should switch or not
         bool switch_to_writes = false;
 
-        if (totalReadQueueSize == 0) {
+        if (totalReadQueueSize == 0 && nvmReadQueueSize == 0) {
```

Comment:
Why this change?

Reply:
This is to make sure that we do not switch to writes only because ReadQueue is empty (nvmReadQeueue can still have some requests)